doc/audit-log/Audit_Log_HLD.md

```diff
@@ -0,0 +1,393 @@
+
+Timestamp at which the login occurred is displayed.
+
+*Console*
+
+Jun 2 22:50:47.510380 sonic INFO login[30983]: pam_unix(login:session): **session closed** for user **admin**
+Jun 2 22:50:47.665222 sonic INFO systemd[1]: Stopped Serial Getty on **ttyS0**
+
+- **Login with invalid username**
+
+Provides invalid user name and IP address of the host invalid user is trying to log in from.
+
+*SSH*
+
+Jun 2 22:51:53.619712 sonic INFO sshd[31688]: **Invalid user** **adminxxx** from **10.14.8.140** port 49090
+
+The message indicates that there was an attempt to connect over ssh with an invalid user name. 
+
+The user name is displayed along with the host IP and port from which user tried to login.
+
+*Console*
+
+Jun 2 22:52:49.080189 sonic NOTICE login[27921]: pam_unix(login:auth): **authentication failure**; logname=LOGIN uid=0 euid=0 **tty=/dev/ttyS0** ruser= rhost=
+Jun 2 22:52:52.055801 sonic NOTICE login[27921]: FAILED LOGIN (1) on '/dev/ttyS0' FOR 'UNKNOWN', Authentication failure
+
+- **Login with invalid password**
+
+Informs that specified user trying to log in with an invalid password and informs IP of the host user is logging in from.
+
+*SSH*
+
+Jun 2 22:53:39.571670 sonic NOTICE sshd[6296]: pam_unix(sshd:auth): **authentication failure**; logname= uid=0 euid=0 tty=ssh ruser= rhost=**10.14.8.140** user=**admin**
+Jun 2 22:53:41.163938 sonic INFO sshd[6296]: Failed password for admin from 10.14.8.140 port 49094 ssh2
+
+The message indicates that user 'admin' entered wrong password that resulted in authentication failure. 
+
+The host IP and port from which user tried the login is displayed in the message.
+
+*Console*
+
+Jun 2 22:54:54.938982 sonic NOTICE login[6927]: pam_unix(login:auth): **authentication failure**; logname=LOGIN uid=0 euid=0 **tty=/dev/ttyS0** ruser= rhost= user=**admin**
+Jun 2 22:54:57.568058 sonic NOTICE login[6927]: FAILED LOGIN (1) on '/dev/ttyS0' FOR 'admin', Authentication failure
+
+- **Session Timeout**
+
+*SSH*
+Jun 10 19:26:32.887528 sonic INFO sshd[24578]: **Timeout, client not responding.**
+
+Jun 10 19:26:33.025597 sonic INFO sshd[24481]: pam_unix(sshd:session): **session closed** for user **admin**
+
+*Console*
+
+Jun 10 20:02:33.904878 sonic INFO systemd[1]: **Stopped Serial Getty** on **ttyS0**.
+
+- **Set Request**
+
+Informs type of request, URI, user name, command string and status of command execution.
+
+*CLI*
+
+Jun 2 22:57:09.060819 sonic INFO mgmt-framework#clish: User "**admin**" command "**tacacs-server key mykey**"  status - **success**
+
+The message displays the command string; its status and name of the user that executed the command.
+
+*REST/gNMI*
+
+Jun 12 19:33:40.728039 sonic INFO mgmt-framework#/usr/sbin/rest_server[711]: [REST-5] User "**admin@10.14.125.28**:55937" request "**PATCH** **/restconf/data/openconfig-system:system/aaa/server-groups/server-group=TACACS/config/openconfig-system-ext:secret-key**" status - **204**
```

Comment:
I think there is no such log in REST server today. Are you planning to introduce new one OR is it constricted by grepping other logs?

Reply:
Adding this message in in handler.go